fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/impl/content/registry/SpreadableBlockRegistryImpl.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import com.google.common.collect.ImmutableMap;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.registry.SpreadableBlockRegistry;
+
+public final class SpreadableBlockRegistryImpl implements SpreadableBlockRegistry {
+	private static final Logger LOGGER = LoggerFactory.getLogger(SpreadableBlockRegistryImpl.class);
+
+	private static final Map<Block, Identifier> SPREADABLE_TYPE;
```

Comment:
This seems to only allow a block to be spreadable. What if I only want one block state of a block to be spreadable?

Reply:
I was told to keep it really simple, so I did.  Then I was told it has to cover a whole bunch more use cases (and here's another)...  Effectively at least for now this is pretty much a dead PR.  I intend to support all this stuff over in Terraform API where there aren't so many constraints on the implementation and I can already use MixinExtras.

Reply:
I understand your frustration, however, once you add something to FAPI, you basically can't make a breaking change at all to ensure compatibility. Therefore, a new API often goes through a lot of deliberation and review. Also, I'm just asking a question about something I noticed.

Reply:
I'm not frustrated, though.  I have an alternative avenue for this.  The only way it impacts me is other mod authors will likely use incompatible mixins because FAPI doesn't provide something.  I don't see this as a case of "diligence" on FAPI's end though.  It's more like a lack of consensus or leadership on what is acceptable for inclusion.  If I had written this just to get it into FAPI then it would be frustrating, but I had to do the work anyway for Terraform API.